### PR TITLE
Fix #1738 - `callback` is not a function when showing "Update Published Project" prompt

### LINKED ISSUE #1738: `callback` is not a function when showing "Update Published Project" prompt
Bug introduced in 392fb9dccc4a228b21dfe88cfe415cb56a6acded

https://github.com/mozilla/thimble.mozilla.org/blob/master/public/editor/scripts/editor/js/fc/publisher.js#L65-L68 <-- these lines use the `showUnpublishedChangesPrompt` function from the `publisher`'s prototype. However, the events that it is attached to send parameters that are not a callback (which is what the function actually expects) and it fails.

We should probably not use function bindings for these event listeners. Instead, let's just use closures like:
```js
bramble.on("event", function() {
  publisher.showUnpublishedChangesPrompt();
});
```

### PR DESCRIPTION
**Pull request by lkisac**
Fix for Issue #1738

### CODE REVIEW

File: public/editor/scripts/editor/js/fc/publisher.js [hunk 1]
```diff
@@ -62,11 +62,19 @@ define(function(require) {
       this.updateDialog(publishUrl, true);
     }
 
-    bramble.on("fileChange", publisher.showUnpublishedChangesPrompt.bind(publisher));
-    bramble.on("fileDelete", publisher.showUnpublishedChangesPrompt.bind(publisher));
-    bramble.on("fileRename", publisher.showUnpublishedChangesPrompt.bind(publisher));
-    bramble.on("folderRename", publisher.showUnpublishedChangesPrompt.bind(publisher));
-
+    bramble.on("fileChange", function() {
+      publisher.showUnpublishedChangesPrompt();
```
Comment (Pomax) [inline public/editor/scripts/editor/js/fc/publisher.js]: remember to also pass all the arguments down in this rewrite:
```
publisher.showUnpublishedChangesPrompt.apply(publisher, arguments);
```
The old code returned a function that would then be called by `bramble` with any number of arguments, so to preserve that functionality we can't just call the function without any args, whatever bramble passes in needs to be passed on to the handler function

Comment (lkisac) [inline public/editor/scripts/editor/js/fc/publisher.js]: Thanks @Pomax, I adjusted it in the latest commit.

**Comment by gideonthomas** [comment]
@lkisac I would revert the change you just made. We don't want to pass in the arguments that the function receives because it does not match the arguments that `showUnpublishedChangesPrompt` accepts. `showUnpublishedChangesPrompt` only accepts a `callback` while the arguments passed in by the `bramble` filechange events are `fileName` and something else. Hence, why this change removes the `bind` which we used before. `apply` basically does the same as `bind` in this case.

File: public/editor/scripts/editor/js/fc/publisher.js (hunk 1 above)
Comment (gideonthomas) [inline public/editor/scripts/editor/js/fc/publisher.js]: > so to preserve that functionality

I think this change is meant to not preserve that functionality as that was what was causing the error :)

**Comment by lkisac** [comment]
@gideonthomas  Thanks.  Good to know.  Please let me know if there is any more changes needed to be made.

**Comment by humphd** [comment]
How close are we to landing this?  I'm getting a lot of confusion from devs who are hitting this and don't know how to overcome it.  It would be great to get the fix up.

**Comment by lkisac** [comment]
@humphd  I couldn't replicate the issues in #1762   I've tested `fileChange`, `fileDelete`, `fileRename`, `folderRename` by changing a file, deleting, renaming, etc. but don't see any error messages.  What else can I try on my end to replicate one of issues devs are running into?

**Comment by gideonthomas** [comment]
@lkisac I don't think there's anything you can do from your end. Sorry for the delay, I'm testing this right now to do one final review.

**Comment by lkisac** [comment]
Ok, thanks @gideonthomas   Will wait for your review.

**Comment by lkisac** [comment]
@gideonthomas Glad it works!

**Pull request merged**